lib/api/request/kuzzleRequest.ts

```diff
@@ -246,6 +246,10 @@ export class KuzzleRequest {
       throw new InternalError('cannot set an error as a request\'s response');
     }
 
+    if (this.context.connection.protocol !== 'http' && result instanceof HttpStream) {
+      throw new InternalError('cannot set an HttpStream as a request\'s response in a non-http protocol');
```

Comment:
You should define a new error in the error manager rather than throwing a raw error